src/net/sourceforge/plantuml/project/lang/ComplementSeveralDays.java

```diff
@@ -30,41 +30,153 @@
  *
  *
  * Original Author:  Arnaud Roques
- * 
+ *
  *
  */
 package net.sourceforge.plantuml.project.lang;
 
-import net.sourceforge.plantuml.command.regex.IRegex;
-import net.sourceforge.plantuml.command.regex.RegexConcat;
-import net.sourceforge.plantuml.command.regex.RegexLeaf;
-import net.sourceforge.plantuml.command.regex.RegexResult;
+import net.sourceforge.plantuml.command.regex.*;
 import net.sourceforge.plantuml.project.Failable;
 import net.sourceforge.plantuml.project.GanttDiagram;
 import net.sourceforge.plantuml.project.Load;
 
+import java.time.Period;
+import java.time.temporal.ChronoUnit;
+import java.util.Locale;
+import java.util.Objects;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static java.time.temporal.ChronoUnit.valueOf;
+
 public class ComplementSeveralDays implements Something {
 
 	public IRegex toRegex(String suffix) {
-		return new RegexConcat( //
-				new RegexLeaf("COMPLEMENT" + suffix, "(\\d+)[%s]+(day|week)s?" + //
-						"(?:[%s]+and[%s]+(\\d+)[%s]+(day|week)s?)?" //
-				)); //
+		return new RegexConcat(
+			new RegexLeaf(
+				getKey(suffix),
+				"(\\d+)[%s]+(day|week|month|quarter|year)s?(?:[%s]+and[%s]+(\\d+)[%s]+(day|week|month|quarter|year)s?)?$"
+			)
+		);
+	}
+
+	public Failable<Load> getMe(GanttDiagram diagram, RegexResult arg, String suffix) {
```

Comment:
I don't think that it would be a valid approach to modify code just to fit the testing, if we can avoid it. Maybe add an interface to the `GanttDiagram`, i.e. `IGanttChart` which would allow us to create mocks via anonymous classes like so?

```java
interface IGanttDiagram {
   int daysInWeek();
}

IGanttDiagram diagram = new IGanttDiagram() {
   @Override
   public int daysInWeek() {
      return workweekLength;
   }
};
```